webapplication/publisher/filters.py

```diff
@@ -23,13 +23,18 @@ class ResultsByACLFilterBackend(filters.BaseFilterBackend):
     def filter_queryset(self, request, queryset, view):
         user_requests = Request.objects.filter(user_id=request.user.id)
         user_requests_list = user_requests.values_list('id', flat=True)
+        user_requests_filter = Q(request_id__in=user_requests_list)
+        is_null_filter = Q(request_id__isnull=True)
+        filtered_queryset = queryset.filter(user_requests_filter)
+        if not filtered_queryset.exists():
+             filtered_queryset = queryset.filter(is_null_filter)
```

Comment:
Didn't get this logic. Should return results generated by user requests and results put by us to the results folder, those will have NULL as request_id.

Reply:
fixed